packages/flame/lib/src/components/mixins/has_ancestor.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+
+/// A mixin that ensures an ancestor is of the given type [T].
+mixin HasAncestor<T extends Component> on Component {
+  /// A reference to an ancestor in the component tree.
+  T get ancestor => _ancestor!;
+  T? _ancestor;
+
+  @override
+  @mustCallSuper
+  void onMount() {
+    if (_ancestor == null) {
+      var c = parent;
+      while (c != null) {
+        if (c is HasAncestor<T>) {
+          _ancestor = c.ancestor;
+          break;
+        } else if (c is T) {
+          _ancestor = c;
+          break;
+        } else {
+          c = c.parent;
+        }
+      }
+    }
+    assert(
+      _ancestor != null,
+      'An ancestor must be of type $T in the component tree',
```

Comment:
The assertion will already throw with a stack trace that relates to the component. Not sure if this would have any benefit directly then? 

If we do this we should do that for the `ParentIsA` and `HasGameRef`.

Reply:
Just a suggestion on a better error message. Rather than searching the trace for the root of the problem, having the runtime type on the message can be helpful. If the case is for consistency, we can improve this and all related messages in a future PR